### PR TITLE
README.md: Specify what Anemone accepts as a double

### PR DESCRIPTION
**Pull request by erikd-ambiata**
The missing double parsing spec.

! @amosr @olorin @nhibberd 
/jury approved @amosr @olorin

### CODE REVIEW

**Comment by amosr** [comment]
👍

File: README.md [hunk 1]
```diff
@@ -27,6 +27,29 @@ getHeaderFile :: ByteString
...
+    nonempty digits
+    [ '.' (nonempty digits) ]      -- Optional decimal point followed by non-empty list of digits
+    [ (`e`|`E`) ['+' | '-'] (nonempty digits) ] -- Optional sign on exponent
+```
+
+The strings specifically not accepted by Anemone include:
+
+* Encodings other than decimal (eg octal or hexadecimal).
+* Numbers that start with the decimal point (eg `.123`).
+* Numbers that end in a decial point (eg `12.`) or have the decimal point immediately preceding the exponent (eg `3.e5`).
```
Comment (olorin) [inline README.md]: Worth mentioning `NaN`/`Inf` here?

**Comment by olorin** [comment]
:cat:

File: README.md (hunk 1 above)
Comment (erikd-ambiata) [inline README.md]: Yes, it is. @amosr Does this handle `Nan`/`Inf`? If so what are the exact strings and are leading `+`/`-` allowed?

Comment (erikd-ambiata) [inline README.md]: Done!

**Pull request merged**